File: app/src/main/java/com/example/fruitsdiary/view/ProgressBarView.java
package com.example.fruitsdiary.view;

import android.content.Context;
import android.util.AttributeSet;
import android.view.LayoutInflater;
import android.view.View;
import android.widget.FrameLayout;
import android.widget.ProgressBar;

import com.example.fruitsdiary.R;

public class ProgressBarView extends FrameLayout {

    private ProgressBar mProgressBar;

    public ProgressBarView(Context context) {
        super(context);
        init(context);
    }

    public ProgressBarView(Context context, AttributeSet attrs) {
        super(context, attrs);
        init(context);
    }

    public ProgressBarView(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
        init(context);
    }

    private void init(Context context){
        View view =
                LayoutInflater.from(context).inflate(R.layout.view_progress_bar, this, true);
        mProgressBar = view.findViewById(R.id.progress_bar);
    }

    public void show(){
        mProgressBar.setVisibility(VISIBLE);
    }

    public void hide(){
        mProgressBar.setVisibility(GONE);
    }

}

File: app/src/main/java/com/example/fruitsdiary/view/ProgressBarView.kt
package com.example.fruitsdiary.view

import android.content.Context
import android.util.AttributeSet
import android.view.LayoutInflater
import android.view.View
import android.widget.FrameLayout
import android.widget.ProgressBar

import com.example.fruitsdiary.R

class ProgressBarView : FrameLayout {

    private lateinit var mProgressBar: ProgressBar

    constructor(context: Context) : super(context) {
        init(context)
    }

    constructor(context: Context, attrs: AttributeSet) : super(context, attrs) {
        init(context)
    }

    constructor(context: Context, attrs: AttributeSet, defStyleAttr: Int) : super(context, attrs, defStyleAttr) {
        init(context)
    }

    private fun init(context: Context) {
        val view = LayoutInflater.from(context).inflate(R.layout.view_progress_bar, this, true)
        mProgressBar = view.findViewById(R.id.progress_bar)
    }

    fun show() {
        mProgressBar.visibility = View.VISIBLE
    }

    fun hide() {
        mProgressBar.visibility = View.GONE
    }

}
